_studio/shared/umc/codec/h264_dec/src/umc_h264_mfx_supplier.cpp

```diff
@@ -1471,11 +1489,21 @@ mfxStatus MFX_Utility::Query(VideoCORE *core, mfxVideoParam *in, mfxVideoParam *
     return sts;
 }
 
-bool MFX_Utility::CheckVideoParam(mfxVideoParam *in, eMFXHWType /* type */)
+bool MFX_Utility::CheckVideoParam(mfxVideoParam *in, eMFXHWType type)
 {
     if (!in)
         return false;
 
+#if defined (MFX_ENABLE_CPLIB)
+    if (in->Protected)
+    {
+        if (type == MFX_HW_UNKNOWN || !IS_PROTECTION_CENC(in->Protected))
+            return false;
+    }
+#else
+    (void)type;
```

Comment:
no need in #else. use unconditionally in the beginning of the function.

Reply:
fixed